src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            elements = new Object[initialCapacity];
+        } else {
+            throw new IllegalArgumentException("Illegal initial capacity " + initialCapacity);
+        }
+    }
```

Comment:
Do you use this constructor? It’s  bad practice to leave unused code in your solutions